### PR TITLE
Address Sebastian's signal comments

### LINKED ISSUE #258: Signals comments
- Maybe mention the difference between `connect` and `connect_local`, and that the properly typed connector functions take care of this distinction themselves. Might be just a footnote
 - `let button = args.get(0).unwrap().get::<Button>().unwrap().unwrap();` explain the `unwrap`s here like for properties, and also why not `args[0]`? It's a slice after all :) Maybe also mention that the first argument to signals is always the object that emitted the signal, and that this specific signal has no other arguments
 - "// Types of the values which will be sent to the receiver" (and below that): Receiver would be better called "signal handler" as that's the GObject terminology
 - You do it correct, but it's worth a note in the `clicked` function that you should really never ever keep a mutex (or refcell) locked/borrowed while: emitting signals, notifying property changes, anything that can cause these to happen. Because that will call into external code, and that external code might call back into your object, trying to use the same mutex/refcell and then you have a problem :)

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Fixes #258

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
>You do it correct, but it's worth a note in the clicked function that you should really never ever keep a mutex (or refcell) locked/borrowed while: emitting signals, notifying property changes, anything that can cause these to happen. Because that will call into external code, and that external code might call back into your object, trying to use the same mutex/refcell and then you have a problem :)

I tried to play a bit, to modify the existing example to nicely demonstrate this problem.
I couldn't reach anything acceptable.

@sdroege maybe you can think of an example, or alternatively a nice explanation using words only  :)

**Comment by sdroege** [comment]
Call something on the object from inside the signal handler that also needs to borrow the state. Like printing the current counter value or something.

**Comment by Hofer-Julian** [comment]
I gave it a try

File: book/src/gobject_signals.md [hunk 1]
```diff
@@ -59,6 +59,44 @@ Whenever we now receive the "max-number-reached" signal, the accompanying number
...
+At `clicked` we can now stop setting the property after we emitted "max-number-reached".
+
+<span class="filename">Filename: listings/gobject_signals/4/custom_button/imp.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/gobject_signals/4/custom_button/imp.rs:button_impl}}
+```
+
+Please also note, that we borrowed our number and bound it to `borrowed_number`.
+While it does not make much sense in our case, it is quite usual to bind locks of Mutexes or borrows of RefCells to variables.
```
Comment (sdroege) [inline book/src/gobject_signals.md:83]: "locks" -> mutex guards?

**Comment by sdroege** [comment]
Sounds good to me otherwise

File: book/src/gobject_signals.md (hunk 1 above)
Comment (Hofer-Julian) [reply book/src/gobject_signals.md]: Fixed

**Comment by Hofer-Julian** [comment]
I think this one can be merged now.
Do you agree @sdroege?

**Review by sdroege — APPROVED** [review]

File: book/listings/gobject_signals/3/custom_button/imp.rs [hunk 2]
```diff
@@ -23,27 +24,71 @@ impl ObjectSubclass for CustomButton {
...
+        match pspec.name() {
+            "number" => {
+                let input_number = value.get().unwrap().unwrap();
+                self.number.replace(input_number);
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn get_property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
```
Comment (bilelmoussaoui) [inline book/listings/gobject_signals/3/custom_button/imp.rs:75]: this needs to be renamed per the latest changes

File: book/listings/gobject_signals/4/custom_button/imp.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        match pspec.name() {
+            "number" => {
+                let input_number = value.get().unwrap().unwrap();
+                self.number.replace(input_number);
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn get_property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
```
Comment (bilelmoussaoui) [inline book/listings/gobject_signals/4/custom_button/imp.rs:75]: Same here

File: book/listings/gobject_signals/3/custom_button/imp.rs (hunk 2 above)
Comment (Hofer-Julian) [reply book/listings/gobject_signals/3/custom_button/imp.rs]: Fixed

File: book/listings/gobject_signals/4/custom_button/imp.rs (hunk 3 above)
Comment (Hofer-Julian) [reply book/listings/gobject_signals/4/custom_button/imp.rs]: Fixed as well :)

**Comment by bilelmoussaoui** [comment]
Thanks Julian!

**Pull request merged**